FinancialTracker/Helpers/Extensions.swift

```diff
@@ -102,6 +102,63 @@ extension UIAlertController {
         alertController.addAction(UIAlertAction(title: "Continue", style: .default, handler: nil))
         return alertController
     }
+    
+    private static func create(basedOnDatabaseError error: DatabaseError?) -> UIAlertController? {
+        if let databaseError = error {
+            let continueAction = UIAlertAction(title: "Continue", style: .default, handler: nil)
+            switch databaseError {
+            case .database(let error):
+                guard let error = error else { return nil }
```

Comment:
I don't like this empty `return nil`, your best option is to not do a guard let here but use a ternary operator later on, like so:

```swift
      case .database(let error):
           let alert = UIAlertController(title: "Database Error", message: error?.localizedDescription ?? "", preferredStyle: .alert)
                 alert.addAction(continueAction)
                 return alert
             case .nonExistingUser:
                 let alert = UIAlertController(title: "Not Existing User", message: error?.localizedDescription ?? "", preferredStyle: .alert)
                 alert.addAction(continueAction)
                 return alert
```

There is really no point in returning `nil` when you are trying to show a helpful alert of what is going on. When you return `nil`, you give no feedback for what happened.

Reply:
Applies to everywhere as usual.

Reply:
Other than that, all the cases here have everything being the same except for just the `title` - what else can you refactor to skip yourself some lines of code?

Reply:
Think